presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TranslateExpressions.java

```diff
@@ -285,4 +513,29 @@ private RowExpression toRowExpression(Expression expression, Rule.Context contex
 
         return SqlToRowExpressionTranslator.translate(expression, types, ImmutableMap.of(), metadata.getFunctionManager(), metadata.getTypeManager(), context.getSession(), false);
     }
+
+    private RowExpression toRowExpression(LambdaExpression expression, Rule.Context context, Map<NodeRef<Expression>, Type> lambdaTypes)
+    {
+        return SqlToRowExpressionTranslator.translate(expression, lambdaTypes, ImmutableMap.of(), metadata.getFunctionManager(), metadata.getTypeManager(), context.getSession(), false);
+    }
+
+    private RowExpression removeOriginalExpression(RowExpression rowExpression, Rule.Context context)
```

Comment:
This function body should be delegated to the one below by having `lambdaTypes` is an empty set